### ISSUE TITLE
Match query params with hash-based routing

### ISSUE THREAD

**Issue by milesfrain** [purescript-contrib/purescript-routing#73]
I don't think this is possible with properly formatted URLs, since the query param should come before the anchor tag.
https://stackoverflow.com/questions/12682952/proper-url-forming-with-query-string-and-anchor-hashtag

Sure, you can get this working with incorrectly-formatted URLs (query after anchor), but other sites are unlikely to present the URL this way (e.g. github OAuth).

Should I add a note about this in the guide?

**Comment by garyb** [comment]
There's nothing wrong with have a query-looking part in the anchor, it's perfectly valid according to RFC 3986, it's just that it won't be part of the actual query, it'll be part of the "fragment" (anchor) instead.

So `/#foo?bar=baz` is a completely valid URI, it's just `?bar=baz` will not be treated as a query string by webserver and such.

**Comment by garyb** [comment]
From the spec:

> A fragment identifier component is indicated by the presence of a
> number sign ("#") character and terminated by the end of the URI.
>
> ```
>    fragment = *( pchar / "/" / "?" )
> ```

So you can see from the ebnf there that `?` is definitely allowed.

**Comment by milesfrain** [comment]
Would it be worth noting the ordering restriction in the guide? For example, we don't want users struggling with trying to figure out how to parse this with hash-based routing:
```
/?bar=baz#foo
```

**Comment by garyb** [comment]
I don't think so... it seems pretty clear to me that hash based routing only deals with things after the `#`?

**Comment by garyb** [comment]
Actually, if it tripped you up, I'm wrong in that assumption, and adding a note certainly wouldn't do any harm.
